storage/cloudspanner/log_storage.go

```diff
@@ -104,13 +106,16 @@ type unsequencedCols struct {
 
 // NewLogStorage initialises and returns a new LogStorage.
 func NewLogStorage(client *spanner.Client) storage.LogStorage {
-	return NewLogStorageWithOpts(client, LogStorageOptions{})
+	return NewLogStorageWithOpts(client, LogStorageOptions{
+		DequeueAcrossMerkleBuckets:              true,
```

Comment:
How strongly do you feel about this?  
Having tested this, I consider the default behavior badly broken and very unexpected. 
The probability of dequing leaves in the default behavior is actually very low. 
1/4 time bucket * 1/16 merkle bucket ~= 64 seconds to dequeue a single leaf. 

Using dequeue across multiple buckets, this drops to 2 seconds, which is the behavior provided by default from mysql. The client libraries depending on the default behavior trigger timeouts all over the place.

Reply:
The problem with this as a default is that the performance will drop dramatically with time - I think what's happening is that Spanner marks "deleted" rows with tombstones and needs a compaction cycle to remove them, so when you're scanning the whole queue to find leaves you also need to scan over all the tombstoned rows too which can be a very large number.  The other "statistical" approach takes longer to integrate, but it stays ~ constant - I'm not totally sure why that is the case, I suspect that it might be something to do with the finer grained keys allow jumping directly into starts of ranges, perhaps avoiding tombstones at the front(?), but the difference in performance is borne out from empirical measurements.

This behaviour is mentioned in some [comments in the code](https://github.com/google/trillian/blob/master/storage/cloudspanner/log_storage.go#L573-L586), but could do with being more clearly documented in a README in this directory.

Reply:
+1 on leaving the old default behaviour intact, as we don't know what performance impact it may bring on MySQL log operators. The O(minutes) integration time might look unexpected, but it's something at least I was aware, and by now the operators probably are as well (I think we have an integration time metric somewhere?).

Reply:
Ah wait, this is`cloudspanner`. For this one, I don't have strong preference :)

Reply:
Got the tests working with the default settings :-)